include/net/nrf_cloud_fota_poll.h

```diff
@@ -79,6 +88,11 @@ struct nrf_cloud_fota_poll_ctx {
 	 * return an error code when an error occurs.
 	 */
 	fota_error_handler_t error_fn;
+
+	/** Optional, user-provided callback function to handle download status.
+	 *  If the function is provided, the user will receive download start and stop events.
+	 */
+	fota_download_handler_t download_status_fn;
```

Comment:
i assume this is intended to be used in the non-blocking case?  if so, i think it would be better to just combine this with the `fota_error_handler_t`, and have just one general fota status callback for all events.

since `fota_error_handler_t` was recently added, there shouldn't be a big impact in changing the API.

Reply:
Good idea, ill do that